internal/middleware/metrics.go

```diff
@@ -0,0 +1,86 @@
+package middleware
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	// MetricCacheHit total number of requests not requiring external lookups
+	MetricCacheHit = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
Now that userdata is has its own, i would include Metadata in the variable name

Reply:
Updated.